Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -2809,7 +2809,8 @@ bool TextureCacheBase::DecodeTextureOnGPU(TCacheEntry* entry, u32 dst_level, con
   if (!info)
     return false;
 
-  const AbstractShader* shader = g_shader_cache->GetTextureDecodingShader(format, palette_format);
+  const AbstractShader* shader = g_shader_cache->GetTextureDecodingShader(
+      format, info->palette_size != 0 ? std::make_optional(palette_format) : std::nullopt);
```

Comment:
No need to do `std::make_optional()` here, you can just pass the value.

Reply:
`info->palette_size != 0 ? palette_format : std::nullopt`

Reply:
[C++'s one-way type inference isn't fancy enough for that](https://gcc.godbolt.org/z/ajjd7ah1T)

Reply:
Ahhh, that makes a lot of sense.  Thank you for the example.